cmd/libsnap-confine-private/mountinfo.c

```diff
@@ -156,67 +156,81 @@ static char *parse_next_string_field(sc_mountinfo_entry * entry,
+			// return NULL. This is an indication of end-of-input
+			// to the caller.
+			if (output_idx == 0) {
+			    return NULL;
+			}
+			// The scanned line is NUL terminated. This ensures that the
+			// terminator is copied to the output buffer.
+			output[output_idx++] = '\0';
+			// NOTE: we are no longer advancing the reading index
+			// because we reached the end of the buffer.
+			break;
+		} else if (c == ' ') {
+			// Fields are space delimited or end-of-string terminated.
+			// Represent either as the end-of-string marker, skip over it,
+			// and stop parsing.
+			output[output_idx++] = '\0';
+			// Advance the reading index by one.
+			input_idx++;
+			// Note that while we are breaking out of the loop the
+			// input_idx variable is used to influence an output
+			// variable. This state is important for subsequent
+			// call where we want to scan the byte immediately
+			// after the space.
 			break;
-		case '\\':
-			// Three *more* octal digits required for the escape sequence.
-			// For reference see mangle_path() in fs/seq_file.c.
-			// Note that is_octal_digit returns false on the string terminator
-			// character NUL and the short-circuiting behavior of && makes
-			// this check correct even if '\\' is the last character of
-			// the string.
+		} else if (c == '\\') {
+			// Three *more* octal digits required for the escape
+			// sequence.  For reference see mangle_path() in
+			// fs/seq_file.c.  Note that is_octal_digit returns
+			// false on the string terminator character NUL and the
+			// short-circuiting behavior of && makes this check
+			// correct even if '\\' is the last character of the
+			// string.
 			s = &input[input_idx];
 			if (is_octal_digit(s[1]) && is_octal_digit(s[2])
 			    && is_octal_digit(s[3])) {
-				// Unescape the octal value encoded in s[1], s[2]
-				// and s[3]. Note that since we have not advanced
-				// input_idx yet, s[0] is the escape character '/'.
-				// Because we are working with byte values there
-				// are no issues related to byte order.
-				output[output_idx] =
+				// Unescape the octal value encoded in s[1],
+				// s[2] and s[3]. Because we are working with
+				// byte values there are no issues related to
+				// byte order.
+				output[output_idx++] =
 				    ((s[1] - '0') << 6) |
 				    ((s[2] - '0') << 3) | ((s[3] - '0'));
-				// Skip over the escaped sequence. Note that
-				// input_idx is automatically advanced in each
-				// loop iteration and that handles the initial
-				// trigger character '/'.
-				input_idx += 3;
+				// Advance the reading index by the length of the escape
+				// sequence.
+				input_idx += 4;
 			} else {
-			    // Partial escape sequence, copy verbatim and
-			    // continue (since we don't use this).
-			    output[output_idx] = c;
+				// Partial escape sequence, copy verbatim and
+				// continue (since we don't use this).
+				output[output_idx++] = c;
+				// Advance the reading index by one.
```

Comment:
Nitpick: this comment is obvious without the reasoning behind it. In this case, probably can just drop it.